tests/System.IO.Pipelines.Tests/ReadableBufferFacts.cs

```diff
@@ -354,7 +354,7 @@ private unsafe void TestValue(ref ReadableBuffer readBuffer, ulong value)
             }
         }
 
-        [Fact]
+        [Fact(Skip = "System.TypeLoadException : A value type containing a by-ref instance field, such as Span<T>, cannot be used as the type for a class instance field.")]
```

Comment:
I think disabling these tests is the wrong approach. In this case, the issue is caused by the fact that the async function is creating a closure around the local span to capture them for re-entering the function when an await returns. This only happens if the span is used on both sides of an await, either directly or indirectly.

We need some documented patterns on things that work and don't work. However, you should be able to re-write these tests in a way that will allow them to work correctly without the error.